lib/datadog/appsec/configuration/settings.rb

```diff
@@ -53,80 +45,80 @@ def self.add_settings!(base)
+                o.type :bool
               end
 
               option :trace_rate_limit do |o|
-                o.default { env_to_int('DD_APPSEC_TRACE_RATE_LIMIT', DEFAULT_APPSEC_TRACE_RATE_LIMIT) } # trace/s
+                o.type :int
+                o.env 'DD_APPSEC_TRACE_RATE_LIMIT' # trace/s
+                o.default 100
               end
 
               option :obfuscator_key_regex do |o|
-                o.default { ENV.fetch('DD_APPSEC_OBFUSCATION_PARAMETER_KEY_REGEXP', DEFAULT_OBFUSCATOR_KEY_REGEX) }
+                o.type :string
+                o.env 'DD_APPSEC_OBFUSCATION_PARAMETER_KEY_REGEXP'
+                o.default DEFAULT_OBFUSCATOR_KEY_REGEX
               end
 
               option :obfuscator_value_regex do |o|
-                o.default do
-                  ENV.fetch(
-                    'DD_APPSEC_OBFUSCATION_PARAMETER_VALUE_REGEXP',
-                    DEFAULT_OBFUSCATOR_VALUE_REGEX
-                  )
-                end
+                o.type :string
+                o.env 'DD_APPSEC_OBFUSCATION_PARAMETER_VALUE_REGEXP'
+                o.default DEFAULT_OBFUSCATOR_VALUE_REGEX
               end
 
               settings :track_user_events do
                 option :enabled do |o|
-                  o.default do
-                    ENV.fetch(
-                      'DD_APPSEC_AUTOMATED_USER_EVENTS_TRACKING',
-                      DEFAULT_APPSEC_AUTOMATED_TRACK_USER_EVENTS_ENABLED
-                    )
-                  end
-                  o.setter do |v|
-                    if v
-                      v.to_s != 'disabled'
-                    else
+                  o.default true
+                  o.type :bool
+                  o.env 'DD_APPSEC_AUTOMATED_USER_EVENTS_TRACKING'
+                  o.env_parser do |env_value|
+                    if env_value == 'disabled'
                       false
+                    else
+                      ['1', 'true'].include?(env_value.strip.downcase)
                     end
                   end
                 end
 
                 option :mode do |o|
-                  o.default do
-                    ENV.fetch('DD_APPSEC_AUTOMATED_USER_EVENTS_TRACKING', DEFAULT_APPSEC_AUTOMATED_TRACK_USER_EVENTS_MODE)
-                  end
+                  o.type :string
+                  o.env 'DD_APPSEC_AUTOMATED_USER_EVENTS_TRACKING'
+                  o.default 'safe'
                   o.setter do |v|
-                    string_value = v.to_s
-                    if APPSEC_VALID_TRACK_USER_EVENTS_MODE.include?(string_value)
-                      string_value
+                    if APPSEC_VALID_TRACK_USER_EVENTS_MODE.include?(v)
+                      v
                     else
                       Datadog.logger.warn(
                         'The appsec.track_user_events.mode value provided is not supported.' \
                         "Supported values are: safe | extended.\n" \
                         'Using default value safe'
```

Comment:
Minor: This was already like this, but since we're here, I suggest removing the newline from the log message, as a lot of logging tools can get tripped with newlines (e.g. show them out of order and whatnot), and thus having it all in a single line avoids any possible confusion.

```suggestion
                        'The appsec.track_user_events.mode value provided is not supported.' \
                        "Supported values are: safe | extended. " \
                        'Using default value safe'
```